### PR TITLE
Timing windows in plot

### PR DESCRIPTION
**Pull request by rpiaggio**
Not sure how to display these. Currently displaying as yellow-ish bands in the chart. We can adjust this.

image

image

### CODE REVIEW

File: explore/src/main/scala/explore/targeteditor/ElevationPlotSection.scala [hunk 1]
```diff
@@ -112,7 +117,17 @@ object ElevationPlotSection:
...
       }
-      .render { (props, ctx, options) =>
+      // FIXME This will be provided by the caller when timing windows are in the ODB
+      .useStreamResourceOnMountBy { (props, ctx, _) =>
+        import ctx.given
+
+        TimingWindowsQuery
+          .query()
+          .map(data => EntryToTimingWindows.get(data.tmpTimingWindows))
+          // _.zoom(TimingWindowsList))
```
Comment (cquiroz) [inline explore/src/main/scala/explore/targeteditor/ElevationPlotSection.scala:127]: Commented code

File: model/shared/src/main/scala/explore/model/TimingWindow.scala [hunk 2]
```diff
@@ -106,6 +110,48 @@ case class TimingWindow private (
 
   def repeatForever: Boolean =
     repetition.exists(_.toOption.exists(_.repeatPeriod.exists(_.repeatFrequency.isEmpty)))
+
+  def toIntervalSeq(within: BoundedInterval[Instant]): IntervalSeq[Instant] = {
```
Comment (cquiroz) [inline model/shared/src/main/scala/explore/model/TimingWindow.scala:114]: I wonder if this expansion logic should go to core, we probably will need it for the scheduler

**Review by cquiroz — APPROVED** [review]
Looks fantastic

File: model/shared/src/main/scala/explore/model/TimingWindow.scala (hunk 2 above)
Comment (rpiaggio) [reply model/shared/src/main/scala/explore/model/TimingWindow.scala:114]: Do we want the model classes in core? In that case the method will go with them.

**Pull request merged**